ui/app/AppLayouts/Wallet/panels/ManageHiddenPanel.qml

```diff
@@ -0,0 +1,227 @@
+import QtQuick 2.15
+import QtQuick.Controls 2.15
+import QtQuick.Layouts 1.15
+
+import StatusQ 0.1
+import StatusQ.Core 0.1
+import StatusQ.Components 0.1
+import StatusQ.Controls 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Models 0.1
+
+import utils 1.0
+import shared.controls 1.0
+
+import AppLayouts.Wallet.controls 1.0
+
+import SortFilterProxyModel 0.2
+
+Control {
+    id: root
+
+    required property var assetsController
+    required property var collectiblesController
+
+    readonly property bool dirty: root.assetsController.dirty || root.collectiblesController.dirty
+    readonly property bool hasSettings: root.assetsController.hasSettings || root.collectiblesController.hasSettings
+
+    background: null
+
+    function clearSettings() {
+        root.assetsController.clearSettings();
+        root.collectiblesController.clearSettings();
+    }
+
+    QtObject {
+        id: d
+
+        property bool assetsExpanded: true
+        property bool collectiblesExpanded: true
+
+        readonly property int assetsCount: root.assetsController.hiddenTokensModel.count + root.assetsController.hiddenCommunityTokenGroupsModel.count
+        readonly property int collectiblesCount: root.collectiblesController.hiddenTokensModel.count + root.collectiblesController.hiddenCommunityTokenGroupsModel.count
+
+        readonly property var filteredHiddenAssets: SortFilterProxyModel {
+            sourceModel: root.assetsController.hiddenTokensModel
+            filters: FastExpressionFilter {
+                expression: {
+                    root.assetsController.hiddenCommunityGroups
+                    return !root.assetsController.hiddenCommunityGroups.includes(model.communityId)
+                }
+                expectedRoles: ["communityId"]
+            }
+        }
+
+        readonly property var filteredHiddenCollectibles: SortFilterProxyModel {
+            sourceModel: root.collectiblesController.hiddenTokensModel
+            filters: FastExpressionFilter {
+                expression: {
+                    root.collectiblesController.hiddenCommunityGroups
+                    return !root.collectiblesController.hiddenCommunityGroups.includes(model.communityId)
+                }
+                expectedRoles: ["communityId"]
+            }
+        }
```

Comment:
I would opt for taking already filtered models in this `ManageHiddenPanel` and therefore avoid dependency on controllers here

Reply:
Not sure if we are thinking about the same thing. What I meant was just moving dependency on controllers one level up and making this component taking exactly what it's visualizing.